patches/bfj+7.1.0.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/bfj/src/match.js b/node_modules/bfj/src/match.js
```

Comment:
Really interesting investigation overall! 
I am not sure if I am the biggest fan of the patch though because patching usually is hard to understand for a reader of the code. The underlying lib might also make changes in the future that would make this workaround break since we are basically relying on an implementation detail.

However, the alternative of permanently pegging to an old version also doesn't seem too appealing (missing out on any updates, esp. security). 
So overall I'd suggest adding some comments explaining the context of the patch to the patch so that later we decrease the friction of understanding it as much as possible. Wdyt?

Reply:
Great suggestion, I added a readme to the patches folder to consolidate the story on this. I am really not a big fan of the patch either :/ but totally agree that from a security/stability perspective it seems like a better solution than pinning to an old version.

Definitely crossing my fingers that a fix makes it into either bfj or jsonpath soon